### ISSUE TITLE
Siwe support

### ISSUE THREAD

**Issue by reslear** [better-auth/better-auth#41]
sample: 
https://docs.reown.com/appkit/vue/core/siwe

**Comment by luksch42** [comment]
Here is the official SIWE proposal. 

ERC-4361: Sign-In with Ethereum
https://eips.ethereum.org/EIPS/eip-4361

**Comment by Bekacru** [comment]
No plans to add SWIE support before v1, but if anyone wants to implement, open for a PR.

**Comment by rokitgg** [comment]
Commenting here for awareness... hope to see a PR for this soon!

**Comment by rajelmn** [comment]
why wanna add SWIE ?

**Comment by rokitgg** [comment]
Many decentralized applications are built upon NextJS and therefore, can get all the benefits of better-auth.

Would be amazing to have a natively-supported SignInWithEthereum plugin for the better-auth ecosystem, since as of now it's very tricky to put these two pieces together. 

A native SIWE plugin would open the room for hundreds of dApps to use Better-Auth for a comfortable auth experience while remaining compliant with the ethos of signing in anonymously with an Ethereum wallet.

**Comment by rokitgg** [comment]
Can confirm that I'm currently working on it 😁

**Comment by 8times4** [comment]
hey @rokitgg did you ever get to finish this? :)

**Comment by rokitgg** [comment]
> hey [@rokitgg](https://github.com/rokitgg) did you ever get to finish this? :)

I do yes, still drafting a PR and writing tests for it.

We were just discussing on Discord about the issue of having the "siwe" npm package as a dependency or implementing it from scratch which is quite a bit of extra work tbh, so will update on this as progress continues

**Comment by 8times4** [comment]
> > hey [@rokitgg](https://github.com/rokitgg?rgh-link-date=2025-05-07T18%3A50%3A36.000Z) did you ever get to finish this? :)
> 
> I do yes, still drafting a PR and writing tests for it.
> 
> We were just discussing on Discord about the issue of having the "siwe" npm package as a dependency or implementing it from scratch which is quite a bit of extra work tbh, so will update on this as progress continues

Awesome!

btw you can replace siwe with viem pretty easily. 
I have made a pugin since that works via viem without any issues.

Lmk if you want the gist for it.

**Comment by rokitgg** [comment]
> > > hey [@rokitgg](https://github.com/rokitgg?rgh-link-date=2025-05-07T18%3A50%3A36.000Z) did you ever get to finish this? :)
> > 
> > 
> > I do yes, still drafting a PR and writing tests for it.
> > We were just discussing on Discord about the issue of having the "siwe" npm package as a dependency or implementing it from scratch which is quite a bit of extra work tbh, so will update on this as progress continues
> 
> Awesome!
> 
> btw you can replace siwe with viem pretty easily. I have made a pugin since that works via viem without any issues.
> 
> Lmk if you want the gist for it.

Yeah I was thinking the same, viem probably does the work better. However viem bundle size is quite a lot when we only need the most basic/core SIWE functions so happy to discuss.

Draft PR is now open so happy to receive any suggestions/contributions for it. Cheers!

**Comment by based64-eth** [comment]
will big tech let us merge this 👁️👁️

**Comment by Kinfe123** [comment]
ref #2579
